core/src/Borrower.sol

```diff
@@ -207,42 +207,46 @@ contract Borrower is IUniswapV3MintCallback {
             liabilities0 -= repayable0;
             liabilities1 -= repayable1;
 
-            if (liabilities0 + liabilities1 == 0 || (liabilities0 > 0 && liabilities1 > 0)) {
-                // If both are zero or neither is zero, there's nothing more to do.
-                // Callbacks/swaps won't help.
-                incentive1 = 0;
-            } else if (liabilities0 > 0) {
-                uint256 unleashTime = slot0_ >> 160;
-                require(0 < unleashTime && unleashTime < block.timestamp, "Aloe: grace");
-
-                liabilities0 /= strain;
-                incentive1 /= strain;
-
-                // NOTE: This value is not constrained to `TOKEN1.balanceOf(address(this))`, so liquidators
-                // are responsible for setting `strain` such that the transfer doesn't revert. This shouldn't
-                // be an issue unless the borrower has already started accruing bad debt.
-                uint256 available1 = mulDiv96(liabilities0, priceX96) + incentive1;
-
-                TOKEN1.safeTransfer(address(callee), available1);
-                callee.swap1For0(data, available1, liabilities0);
+            // Decide whether to swap or not
+            bool shouldSwap;
+            assembly ("memory-safe") {
+                // If both are zero or neither is zero, there's nothing more to do
+                shouldSwap := xor(gt(liabilities0, 0), gt(liabilities1, 0))
+                // Divide by `strain` and check again. This second check can generate false positives in cases
+                // where one division (not both) floors to 0, which is why we `and()` with the check above.
+                liabilities0 := div(liabilities0, strain)
+                liabilities1 := div(liabilities1, strain)
+                shouldSwap := and(shouldSwap, xor(gt(liabilities0, 0), gt(liabilities1, 0)))
```

Comment:
👍 If `liabilities0` or `liabilities1` gets truncated as a result of `/= strain`, then we won't swap. The liquidator would probably have to adjust the `strain` in order to perform the liquidation successfully.